pkg/oc/cli/cmd/rollout/cancel.go

```diff
@@ -98,27 +116,38 @@ func (o *CancelOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, out io.
 		return err
 	}
 
-	r := f.NewBuilder().
+	o.Printer = func(successMsg string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.Complete(successMsg)
+		return o.PrintFlags.ToPrinter()
+	}
+
+	o.Builder = f.NewBuilder()
+	o.Resources = args
+	return err
+}
+
+func (o CancelOptions) Run() error {
+	r := o.Builder.
 		WithScheme(ocscheme.ReadingInternalScheme).
```

Comment:
We need to always set decoding versions: `WithScheme(ocscheme.ReadingInternalScheme, ocscheme.ReadingInternalScheme.PrioritizedVersionsAllGroups()...)` otherwise you might end up with internals.